### PR TITLE
lint for noarch python without a bound on python version

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
                 " form. See lines %s." % (bad_lines,)
             )
 
+    # 25: require a lower bound on python version
+    if build_section.get("noarch") == "python" and not outputs_section:
+        for req in run_reqs:
+            if req.startswith("python") and req != "python":
```
Comment (mariusvniekerk) [inline conda_smithy/lint_recipe.py:521]: This should make it work for things like `python-foo` as well

```suggestion
            if (req.strip().split()[0] == "python") and (req != "python"):
```

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
 
+    # 25: require a lower bound on python version
+    if build_section.get("noarch") == "python" and not outputs_section:
+        for req in run_reqs:
+            if (req.strip().split()[0] == "python") and (req != "python"):
+                break
+        else:
+            lints.append(
+                "noarch: python recipes are recommended to have a lower bound "
+                "on minimum python bound. This recommendation will become "
```
Comment (ocefpaf) [inline conda_smithy/lint_recipe.py:526]: Maybe "a minimum python bound" only is clearer then "a lower bound on minimum python bound"?

**Review by ocefpaf — APPROVED** [review]

File: conda_smithy/lint_recipe.py (hunk 2 above)
Comment (beckermr) [inline conda_smithy/lint_recipe.py:526]:
```suggestion
                "on the python version. This recommendation will become "
```

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
 
+    # 25: require a lower bound on python version
+    if build_section.get("noarch") == "python" and not outputs_section:
+        for req in run_reqs:
+            if (req.strip().split()[0] == "python") and (req != "python"):
+                break
+        else:
+            lints.append(
+                "noarch: python recipes are recommended to have a lower bound "
+                "on the python version. This recommendation will become "
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:526]:
```suggestion
                "on the python version. This recommendation will become a "
```

File: tests/test_lint_recipe.py [hunk 4]
```diff
@@ -580,6 +580,54 @@ def test_bad_requirements_order(self):
         lints, hints = linter.lintify(meta)
         self.assertNotIn(expected_message, lints)
 
+    def test_noarch_python_bound(self):
+        expected_message = (
+            "noarch: python recipes are recommended to have a lower bound "
+            "on the python version. This recommendation will become "
```
Comment (isuruf) [inline tests/test_lint_recipe.py:586]:
```suggestion
            "on the python version. This recommendation will become a "
```

File: conda_smithy/lint_recipe.py [hunk 5]
```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
+    # 25: require a lower bound on python version
+    if build_section.get("noarch") == "python" and not outputs_section:
+        for req in run_reqs:
+            if (req.strip().split()[0] == "python") and (req != "python"):
+                break
+        else:
+            lints.append(
+                "noarch: python recipes are recommended to have a lower bound "
+                "on the python version. This recommendation will become a "
+                "requirement in a future version."
```
Comment (beckermr) [inline conda_smithy/lint_recipe.py:527]:
```suggestion
                "requirement in the future."
```

File: tests/test_lint_recipe.py [hunk 6]
```diff
@@ -580,6 +580,54 @@ def test_bad_requirements_order(self):
         lints, hints = linter.lintify(meta)
         self.assertNotIn(expected_message, lints)
 
+    def test_noarch_python_bound(self):
+        expected_message = (
+            "noarch: python recipes are recommended to have a lower bound "
+            "on the python version. This recommendation will become a "
+            "requirement in a future version."
```
Comment (beckermr) [inline tests/test_lint_recipe.py:587]:
```suggestion
            "requirement in the future."
```

**Review by beckermr — APPROVED** [review]

**Pull request merged**